File: openai/clients/node/test-connection.js
/**
 * Test OpenAI API connection + core capabilities
 */
const { OpenAIClient } = require('./openai-client');

async function main() {
  console.log('\n🔌 OpenAI Connection Test\n');

  const client = new OpenAIClient();
  let passed = 0, failed = 0;

  async function test(name, fn) {
    try {
      await fn();
      console.log(`  ✅ ${name}`);
      passed++;
    } catch (e) {
      console.log(`  ❌ ${name}: ${e.message || JSON.stringify(e)}`);
      failed++;
    }
  }

  await test('Chat completion (gpt-4o-mini)', async () => {
    const res = await client.generateContent({
      prompt: 'Say "OpenAI connected" — nothing else.',
      model: 'gpt-4o-mini',
      max_tokens: 10
    });
    if (!res.text.toLowerCase().includes('openai')) throw new Error(`Got: ${res.text}`);
  });

  await test('Streaming (gpt-4o-mini)', async () => {
    let text = '';
    for await (const chunk of client.generateContentStream({
      prompt: 'Count to 3.', model: 'gpt-4o-mini', max_tokens: 20
    })) {
      text += chunk.text;
    }
    if (!text) throw new Error('No streaming output');
  });

  await test('Embeddings (text-embedding-3-small)', async () => {
    const emb = await client.embedContent('SkyyRose luxury fashion');
    if (!Array.isArray(emb) || emb.length < 100) throw new Error('Bad embedding');
  });

  console.log(`\n  ${passed} passed · ${failed} failed`);
  process.exit(failed > 0 ? 1 : 0);
}

main().catch(e => { console.error(e); process.exit(1); });

File: openai/clients/python/test_connection.py
"""Test OpenAI Python client connection"""
from openai_client import OpenAIClient

def main():
    print('\n🔌 OpenAI Python Connection Test\n')
    client = OpenAIClient()
    passed = failed = 0

    def test(name, fn):
        nonlocal passed, failed
        try:
            fn()
            print(f'  ✅ {name}')
            passed += 1
        except Exception as e:
            print(f'  ❌ {name}: {e}')
            failed += 1

    def t_chat():
        res = client.generate_content('Say "Python connected" — nothing else.', model='gpt-4o-mini')
        assert 'python' in res['text'].lower() or 'connected' in res['text'].lower()

    def t_embed():
        emb = client.embed_content('SkyyRose luxury fashion')
        assert isinstance(emb, list) and len(emb) > 100

    test('Chat completion (gpt-4o-mini)', t_chat)
    test('Embeddings (text-embedding-3-small)', t_embed)

    print(f'\n  {passed} passed · {failed} failed')
    exit(1 if failed else 0)

if __name__ == '__main__':
    main()
